source/solvers/vof.cc

```diff
@@ -2095,20 +2201,49 @@ VolumeOfFluid<dim>::apply_peeling_wetting(const unsigned int i_bc,
                 round(phase_lighter_fluid_q / n_q_points_face);
               double phase_values_cell =
                 phase_values_q / static_cast<double>(n_q_points_face);
+              double pressure_values_cell =
+                pressure_values_q / static_cast<double>(n_q_points_face);
+
+              // Check peeling condition on the pressure gradient
+              if (pressure_values_cell < this->pressure_monitored_avg and
+                  nb_pressure_grad_meet_peel_condition > n_q_points_face / 2)
+                {
+                  // Peel the higher density fluid
+                  // conditions phase_values_cell < 1 or phase_values_cell
+                  // > 0 prevent from treating values outside of the range
+                  // [0,1]
+                  if (phase_denser_fluid_cell == 1 and phase_values_cell > 0.5)
+                    {
+                      // Progressive phase value change
+                      double new_phase =
+                        std::max(phase_values_cell - 0.25,
```

Comment:
What is this 0.25?

Reply:
this number is a bit arbitrary, I use it to change progressively the phase value rather than go from `1` to `0` directly. Would you prefer it to be a parameter?

Reply:
Let's not add parameter. This mechanism is still very project specific anyway...